### PR TITLE
[RA-2131] add method set keepalive timeout

### PR DESCRIPTION
**Pull request by amatyushov**

### CODE REVIEW

File: lib/cpp/src/thrift/transport/TSocket.cpp [hunk 1]
```diff
@@ -780,6 +780,39 @@ void TSocket::setKeepAlive(bool keepAlive) {
   }
 }
 
+void TSocket::setTimeoutKeepAlive(int keepalive_time, int keepalive_interval) {
+
+  if (socket_ == THRIFT_INVALID_SOCKET) {
+    return;
+  }
+  if (!keepAlive_) {
```
Comment (AnatolyZ) [inline lib/cpp/src/thrift/transport/TSocket.cpp]: Why do we need keepAlive_ turned on?

File: lib/cpp/src/thrift/transport/TSocket.cpp [hunk 2]
```diff
@@ -780,6 +780,39 @@ void TSocket::setKeepAlive(bool keepAlive) {
...
+  // The keepaliveinterval member specifies the interval, in milliseconds
+  // between when successive keep-alive packets are sent if no acknowledgement is received   
+  vals.keepaliveinterval = keepalive_interval; // milliseconds
+
+  DWORD numBytesReturned = 0;   // not really used AFAICT
+
+  int ret = WSAIoctl(socket_, SIO_KEEPALIVE_VALS, &vals, sizeof(vals), nullptr, 0,
+                     &numBytesReturned, nullptr, nullptr);
+
+  if (ret == -1) {
```
Comment (AnatolyZ) [inline lib/cpp/src/thrift/transport/TSocket.cpp]: Maybe return result from function?

File: lib/cpp/src/thrift/transport/TSocket.cpp [hunk 3]
```diff
@@ -780,6 +780,39 @@ void TSocket::setKeepAlive(bool keepAlive) {
...
+void TSocket::setTimeoutKeepAlive(int keepalive_time, int keepalive_interval) {
+
+  if (socket_ == THRIFT_INVALID_SOCKET) {
+    return;
+  }
+  if (!keepAlive_) {
+    return;
+  }
+
+  struct tcp_keepalive vals;
```
Comment (AnatolyZ) [inline lib/cpp/src/thrift/transport/TSocket.cpp]: maybe init struct with {} ?

File: lib/cpp/src/thrift/transport/TSocket.cpp (hunk 1 above)
Comment (amatyushov) [reply lib/cpp/src/thrift/transport/TSocket.cpp]: I removed this condition

**Review by akhvoshch — APPROVED** [review]

**Review by IgorIonov — APPROVED** [review]

**Pull request merged**